packages/patternfly-4/react-table/src/components/Table/Table.js

```diff
@@ -139,9 +139,29 @@ class Table extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      headerData: []
+      headerData: [],
+      allRowsSelected: props.onSelect ? this.areAllRowsSelected(props.rows) : false
     };
+    this.isSelected = this.isSelected.bind(this);
+    this.areAllRowsSelected = this.areAllRowsSelected.bind(this);
   }
+
+  componentDidUpdate(prevProps) {
```

Comment:
I don't really like `componentDidUpdate` function, it could lead to infinite loop (because of calling set state in it's body). Might be better to use [static-getderivedstatefromprops](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops).

Reply:
After further investigation, I don't think we need to store `allRowsSelected` in state. Rather send
```JS
const headerData = calculateColumns(cells, {
 // ...
 allRowsSelected: this.props.onSelect ? this.areAllRowsSelected(this.props.rows) : false
})
```

Reply:
> I don't really like `componentDidUpdate` function, it could lead to infinite loop

It does have a conditional check

Reply:
Fixed